# NHANES Data Analysis & Obesity Prediction
## 03 — EDA Part 1: Descriptive Statistics

## This notebook summarizes the key demographic and clinical variables in the cleaned NHANES dataset. Both continuous and categorical variables are explored to understand the distribution of the adult population included in the study.

## 1. Load the Cleaned Dataset

In [ ]:
import pandas as pd

# Load the dataset
df = pd.read_csv("\data\clean_merged_data.csv")

## 2. Descriptive Statistics for Continuous Variables

In [2]:
# Describe continuous variables
num_vars = ["RIDAGEYR", "INDFMPIR", "BMXWAIST", "BMXARMC", "BMXHIP", "BMXHT"]
desc_stat = df[num_vars].describe().T
print(desc_stat)
desc_stat.to_csv("descriptive_stats.csv")

            count        mean        std           min    25%     50%     75%  \
RIDAGEYR  15025.0   50.539168  18.373780  1.800000e+01   35.0   52.00   66.00   
INDFMPIR  15025.0    2.638108   1.543417  5.397605e-79    1.4    2.19    4.07   
BMXWAIST  15025.0  100.076000  17.012216  5.640000e+01   88.6   98.10  110.40   
BMXARMC   15025.0   33.490369   5.313977  1.600000e+01   29.9   33.00   36.70   
BMXHIP    15025.0  107.210050  14.320727  7.400000e+01   97.6  104.00  114.00   
BMXHT     15025.0  166.828426  10.043824  1.311000e+02  159.4  166.40  174.10   

            max  
RIDAGEYR   80.0  
INDFMPIR    5.0  
BMXWAIST  187.5  
BMXARMC    64.5  
BMXHIP    187.5  
BMXHT     200.7  


## Interpretation

###  The average age (RIDAGEYR) is around 50 years, with a wide range from 18 to 80, showing a diverse adult population.

###  The mean income ratio (INDFMPIR) is 2.64, with values ranging from very low to a maximum of 5, indicating socioeconomic diversity.

### - Body measurements (waist, arm, hip, height):  
###  These variables show expected variation in an adult population.

###  - Waist circumference averages 100 cm

###  - Hip circumference averages 107 cm

###  - Height averages 167 cm

### These values align with typical U.S. adult anthropometric patterns.

## 3. Descriptive Statistics for Categorical Variables

In [3]:
# Describe categorical variables
mappings = {
 "RIAGENDR": {1.0: "Male", 2.0: "Female"},
 "DMDMARTZ": {1.0: "Married", 2.0: "Widowed", 3.0: "Divorced/Separated", 77.0: "Refused", 99.0: "Don't know"},

 "RIDRETH3": {1.0: "Mexican American", 2.0: "Other Hispanic", 3.0: "Non-Hispanic White", 4.0: "Non-Hispanic Black",
    6.0: "Non-Hispanic Asian", 7.0: "Other/Multi-Race"},

 "DMDEDUC2": {1.0: "Less than 9th grade", 2.0: "9–11th grade", 3.0: "High school/GED", 4.0: "Some college/AA",
    5.0: "College graduate or above", 7.0: "Refused", 9.0: "Don't know"},
 
 "Obese": {0.0: "Not Obese", 1.0: "Obese"}}

df_labeled = df.copy()
for col, mapping in mappings.items():
    df_labeled[col] = df_labeled[col].map(mapping)

cat_vars = list(mappings.keys())

def count_percent(series):
    counts = series.value_counts(dropna=False)
    percentages = series.value_counts(normalize=True, dropna=False) * 100
    return pd.DataFrame({"Count": counts, "Percentage": percentages.round(2)})

cat_summary = {col: count_percent(df_labeled[col]) for col in cat_vars}
all_summary = pd.concat(cat_summary, axis=0)
print(all_summary)

                                    Count  Percentage
RIAGENDR Female                      7937       52.83
         Male                        7088       47.17
DMDMARTZ Married                     8779       58.43
         Widowed                     3379       22.49
         Divorced/Separated          2855       19.00
         Refused                        7        0.05
         Don't know                     5        0.03
RIDRETH3 Non-Hispanic White          6665       44.36
         Non-Hispanic Black          3087       20.55
         Other Hispanic              1538       10.24
         Mexican American            1478        9.84
         Non-Hispanic Asian          1419        9.44
         Other/Multi-Race             838        5.58
DMDEDUC2 Some college/AA             5229       34.80
         College graduate or above   4182       27.83
         High school/GED             3293       21.92
         9–11th grade                1385        9.22
         Less than 9th grade

## Interpretation
###  The sample is slightly female‑dominant.
###  Most adults are married, with a notable proportion widowed or divorced.
###  The dataset reflects a diverse population, with large Non‑Hispanic White, Black, and Hispanic groups.
###  Most participants completed high school or some college, with a smaller share holding a college degree.
###  About two‑fifths of adults meet the BMI criteria for obesity.

## 4. Notebook Summary

### This notebook provides a clear overview of the NHANES adult dataset by:

### - Summarizing key continuous variables (age, income, body measurements)

### - Converting coded categorical variables into meaningful labels

### - Calculating counts and percentages for demographic and clinical categories

### - Highlighting the distribution of obesity in the population

### These descriptive statistics form the foundation for deeper analysis, visualization, and modeling in the next steps of the project.